### PR TITLE
DEVPROD-7340 Move ConditionalWrapper to lib and setup vitest for jsdom testing in lib

### PR DESCRIPTION
**Pull request by khelif96**
DEVPROD-7340

### Description
This is the first part in a series of PRs in order to help breakup the setup work required to move the LG table component over to the lib package.

Move the ConditionalWrapper component to lib as well as initialize the testing infrastructure in the lib package to support jsdom based matchers.

### CODE REVIEW

File: packages/lib/config/vitest/setupTests.ts
Comment (sophstad) [inline packages/lib/config/vitest/setupTests.ts:1]: This needs to be referenced in `vitest.config.ts` to take effect

File: packages/lib/config/vitest/global-setup.ts
Comment (sophstad) [inline packages/lib/config/vitest/global-setup.ts:1]: Same as above

**Review by sophstad — CHANGES_REQUESTED** [review]

File: packages/lib/config/vitest/setupTests.ts
Comment (khelif96) [reply packages/lib/config/vitest/setupTests.ts:1]: Thanks!

File: packages/lib/config/vitest/setupTests.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import "vitest-canvas-mock";
+
+// @ts-ignore
+window.crypto.randomUUID = (() => {
+  let value = 0;
+  return () => {
+    value += 1;
+    return value.toString();
+  };
+})();
```
Comment (minnakt) [inline packages/lib/config/vitest/setupTests.ts:12]: is this block necessary? If not, could we delete it, and if it is, can we leave a comment about why it's necessary?

**Pull request merged**